File: languages/HamsterSimulatorLanguage.sandbox/source_gen/HamsterSimulatorLanguage/sandbox/HamsterLibrary/Row.cpp
#include "Row.h"
#include <memory>
#include <alf/library/collectionfunctions/CollectionFunctions.hpp>
#include "Cell.h"
#include <optional>
#include <vector>

namespace HamsterLibrary
{
  Row::Row(int columns)
  {
    for (int columnIndex = 1; columnIndex <= columns; columnIndex++)
    {
      std::shared_ptr<Cell> cell = std::make_shared<Cell>();
      if (columnIndex > 1)
      {
        std::optional<std::shared_ptr<Cell>> leftCell = alf::library::primitivebehaviors::CollectionFunctions::at(this->cells, columnIndex - 1 - 1);
        cell->west = leftCell;
        leftCell.value()->east = cell;
      }

      this->cells.push_back(cell);
    }
  }
}

File: languages/HamsterSimulatorLanguage.sandbox/source_gen/HamsterSimulatorLanguage/sandbox/HamsterLibrary/Row.cs
/// <filename>
///     Row.cs
/// </filename>
using Alf.Library.PrimitiveBehaviors;
using HamsterLibrary;
using System.Collections.Generic;

namespace HamsterLibrary
{
    public class Row
    {

        public Row(int columns)
        {
            for (int columnIndex = 1; columnIndex <= columns; columnIndex++)
            {
                HamsterLibrary.Cell cell = new HamsterLibrary.Cell();
                if (columnIndex > 1)
                {
                    HamsterLibrary.Cell leftCell = this.cells.At(columnIndex - 1 - 1);
                    cell.west = leftCell;
                    leftCell.east = cell;
                }

                this.cells.Add(cell);
            }
        }

        public System.Collections.Generic.List<HamsterLibrary.Cell> cells = new System.Collections.Generic.List<HamsterLibrary.Cell>();
    }
}
